src/libAtomVM/term.h

```diff
@@ -1334,30 +1346,19 @@ static inline int term_list_length(term t, int *proper)
  * @param b second term
  * @return 1 if they are the same, 0 otherwise.
  */
-static inline int term_exactly_equals(term a, term b, Context *ctx)
+static inline int term_exactly_equals(term a, term b, GlobalContext *global)
 {
     if (a == b) {
         return 1;
     } else {
-        return term_compare(a, b, ctx) == 0;
-    }
-}
-
-/**
- * @brief Returns 1 if given terms are equal.
- *
- * @details Compares 2 given terms and returns 1 if they are the same or they have same numeric value.
- * @param a first term
- * @param b second term
- * @return 1 if they are the same, 0 otherwise.
- */
-static inline int term_equals(term a, term b, Context *ctx)
-{
-    if (a == b) {
-        return 1;
-    } else {
-        //TODO: add parameter for exactly equals.
-        return term_compare(a, b, ctx) == 0;
+        TermCompareResult result = term_compare(a, b, global);
+        if (result == TermEquals) {
+            return 1;
+        } else if (UNLIKELY(result == TermCompareMemoryAllocFail)) {
+            abort();
```

Comment:
This should use the AVM_ABORT macro, so that a module and line number is recorded (if applicable)